### PR TITLE
Fix backendLdLibraryPath

### PR DESCRIPTION
**Pull request by mulderr**
### Pull Request Description
Currently `backendLdLibraryPath` depends on CWD. It should instead reuse `absAppDir`.

If you build luna studio and start it from the `dist` directory then everything works fine. If however you start it from any other directory:
- some backend services fail to start as they cannot find zeromq libs: `exited: luna-double-representation (exit status 127; not expected)`
- atom starts but projects do not fully load because the backend isn't properly initialized (super confusing if the user doesn't notice errors in the console)

This PR fixes those issues.

### Important Notes
Nothing to report.

### Checklist

- [x] The documentation has been updated if necessary.
- [x] The code conforms to the [Haskell Style Guide](https://github.com/luna/luna/blob/master/doc/haskell-style-guide.md) and/or the [TypeScript Style Guide](https://github.com/luna/luna-studio/blob/master/doc/typescript-style-guide.md).
- [x] The code has been tested where possible.

### CODE REVIEW

File: runner/src/StudioRunner.hs [hunk 1]
```diff
@@ -143,6 +144,7 @@ absAppDir = do
                  else parent . parent . parent -- drop bin/public/luna-studio/
     pure $ stepUp runnerDir
 
+-- | Absolute path to application resource directory inside user home directory
```
Comment (mwu-tow) [inline runner/src/StudioRunner.hs]: I'd avoid "application resource directory" phrase, as we have `_resourcesFolder` that means something else. I'd go with something more like "application data directory" or just "application directory" (well, all directories contain data :P ).

File: runner/src/StudioRunner.hs [hunk 2]
```diff
@@ -130,6 +130,7 @@ getRunnerDir = liftIO $ do
     x <- getExecutablePath
     parent <$> parseAbsFile x
 
+-- | Absolute path to application installation directory
```
Comment (mwu-tow) [inline runner/src/StudioRunner.hs]: I wouldn't call this necessarily installation directory — IIRC on Linux it resolves to mounted appimage fs, not the place where appimage was installed.
I'd go more with something like "application package root".

Also, total nitpick, but: we don't need to write in comment that this is absolute path, if we have this explicitly coded both in the type signature and the function name.

File: runner/src/StudioRunner.hs [hunk 3]
```diff
@@ -209,7 +211,7 @@ userStudioAtomHome = do
     baseDir   <- buildDirectoryPath absHomeDir [configHomeFolder, appName] >>= \x -> fmap (x </>) (parseRelDir =<< version)
     pure $ baseDir </> (runnerCfg ^. studioHome)
 backendLdLibraryPath = do
-    ldLibPath <- PIO.getCurrentDir
+    ldLibPath <- absAppDir
```
Comment (mwu-tow) [inline runner/src/StudioRunner.hs]: While we are at this, let's fix the naming as well. Obviously application directory is not a `ldLibPath` — the returned value from function is.

**Comment by mulderr** [comment]
Any news on this? Always happy to make more changes.

**Pull request merged**